cmd/otel-agent/main.go

```diff
@@ -17,7 +17,13 @@ import (
 	_ "github.com/DataDog/datadog-agent/pkg/version"
 )
 
+var BYOC string
+
 func main() {
+	var BYOCBool bool
```

Comment:
nit: I'd maybe put this logic inside `MakeRootCommand()` instead of here after moving the variable as suggested above. 

We have to be resilient to case, so I would make a case a insensitive comparison using. `strings.EqualFold(BYOC, "true")`